### ISSUE TITLE
support shared ipad/network users

### ISSUE THREAD

**Issue by groob** [micromdm/micromdm#285]
As of #283

When a new managed user is created, the existing managed users from the device are deleted. This is because there can only be one managed user per macOS device so we clean up the existing managed user from the device when adding a new one. 

If you have access to shared ipads or an environment with network accounts and need this to work, ping me.

**Comment by xndr** [comment]
pinging @groob 
It doesn't look like activating ipads in shared mode is supported yet in micromdm? You asked users to ping you about this?

**Comment by jessepeterson** [comment]
@xndr Shared iPads have a funky way of encoding the `UserID` in each MDM request. I don't think MicroMDM supports that. However, [NanoMDM](https://github.com/micromdm/nanomdm) has that special handling included (though, I haven't had a chance to test it on real shared ipad devices yet).

MicroMDM has very limited support for `UserAuthenticate` messages (similar to NanoMDM) which are how "network accounts" MDM are handled on macOS — though we simply reject all UserAccounts in MicroMDM, I think.

Is this something you're looking for?

**Comment by xndr** [comment]
> Is this something you're looking for?

I'm currently using micromdm for macs and ipads, but it is activating the iPads in single-user supervised mode. I would like to activate them in shared mode, to avoid having to manually activate the iPads with Configurator2's "Enable Shared iPad Mode".

MicroMDM is unaware of our user accounts.

**Comment by RoMaTiX99** [comment]
@xndr / @jessepeterson I‘m looking for a way to make my iPad Pro Shared User ready. According to apple docs it must be a managed device, but doesn’t need to be a school account anymore. 

Is there any update on this? Did anyone managed to get this working or found another workaround?

**Comment by xndr** [comment]
Yes, you can do it with a non-school device. You still need to have an ABM account as only managed Apple IDs can login. At least, it was so in my experience.

**Comment by jessepeterson** [comment]
I'm not sure MicroMDM will support a Shared iPad. The user enrollment identifier style is different than a normal user id. That said I've never tested it.

However, there is code in NanoMDM for that specific enrollment style (and targeting the specific Shared iPad user for MDM commands), though it hasn't been tested, either: https://github.com/micromdm/nanomdm/blob/main/docs/operations-guide.md#enrollment-ids

**Comment by RoMaTiX99** [comment]
Thank you for your help. It seems like @xndr is right and you still need to have a paided ABM Account. Sadly then it’s not worth it for me.

**Comment by xndr** [comment]
> Thank you for your help. It seems like @xndr is right and you still need to have a paided ABM Account. Sadly then it’s not worth it for me. 

You don't need to pay for anything, but when you activate an iPad in Shared Mode, only the managed Apple IDs of the organization that activated the device can login.

ABM does not cost anything, but usually Apple wants a company name and a DUNS number to register for ABM.
